src/main/java/io/vertx/core/eventbus/EventBusOptions.java

```diff
@@ -647,7 +647,7 @@ public int getClusterPublicPort() {
    * @return a reference to this, so the API can be used fluently
    */
   public EventBusOptions setClusterPublicPort(int clusterPublicPort) {
-    if (clusterPublicPort < 0 || clusterPublicPort > 65535) {
+    if (clusterPublicPort != DEFAULT_CLUSTER_PUBLIC_PORT && (clusterPublicPort < 0 || clusterPublicPort > 65535)) {
```

Comment:
I think actually a better fix would be to update the `toJson()` method and remove the `clusterPublicPort` when it is `-1` so we don't end up in this case

Reply:
OK,that makes sense and looks better :)